### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by jonathan016**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/enQtODAwOTg0OTUzNDEwLTdlY2M4NmNlZTMzM2Q4MDUyZThkYzQwNmE2MDVhNTRiZDcxNTNlMWJhNTZkNWE1OGU2Nzg3MmY1OWQzN2Q5Mzk) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
Fixes some of #44 and contains improved documentation on the following classes:
- [x] `Convolution`
- [x] `Conv1D`
- [x] `Conv2D`
- [x] `Conv3D`
- [x] `BatchNorm`
- [x] `Dropout`

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (see comments below)
- [x] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
- [x] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] `Convolution` - Added documentation on how convolution works, benefits and drawbacks, and paper references.
- [x] `Conv1D` - Specified how `Conv1D` differs from `Convolution` by its dimension size and common use cases.
- [x] `Conv2D` - Specified `Conv2D` specifications and common use cases.
- [x] `Conv3D` - Specified `Conv3D` specifications, how it differs from others, benefits/drawbacks, and common use cases.
- [x] `BatchNorm` - Explained the concept and formal definition of batch normalization, its proposal paper, and benefits/drawbacks.
- [x] `Dropout` - Explained the concept of dropout, its proposal paper, and benefits/drawbacks.

## Comments ##
- For `Conv1D`, `Conv2D`, and `Conv3D`, I was quite unsure of what else to include since these three share common operations as specified in `Convolution` class, so if anything is missing please let me know so the missing aspects can be obtained.
- `Dropout` has different papers, some suggesting that this [2014 paper](https://www.cs.toronto.edu/~hinton/absps/JMLRdropout.pdf) is the original proposal while others suggest the earlier [2012 paper](https://arxiv.org/abs/1207.0580) as the original proposal, despite the former having more citations. I need further confirmation on this.

### CODE REVIEW

**Comment by zachgk** [comment]
For the dropout, it might be that the first one was arxiv only while the second one was submitted to a journal/conference (JMLR), so either one would work fine. I would have also gone with the second one since it seems to be more cited and a bit more thorough. The main reason we want to have the papers is so users who are more interested in learning the theory and the details behind techniques know where to look, so having a slightly more developed paper for them to look at would help them more

**Comment by jonathan016** [comment]
Alright then, thank you so much @zachgk!
